core/src/main/java/com/predic8/membrane/core/lang/ScriptingUtils.java

```
@@ -0,0 +1,70 @@
+ */
+
+package com.predic8.membrane.core.lang;
+
+import com.fasterxml.jackson.databind.*;
+import com.predic8.membrane.core.exchange.*;
+import com.predic8.membrane.core.http.*;
+import com.predic8.membrane.core.interceptor.*;
+import com.predic8.membrane.core.util.*;
+import org.slf4j.*;
+
+import java.util.*;
+
+import static com.predic8.membrane.core.interceptor.Interceptor.Flow.REQUEST;
+import static com.predic8.membrane.core.util.FileUtil.readInputStream;
+
+public class ScriptingUtils {
+
+    private static final Logger log = LoggerFactory.getLogger(ScriptingUtils.class.getName());
+
+    private static final ObjectMapper om = new ObjectMapper();
+
+    public static HashMap<String, Object> createParameterBindings(Exchange exc, Message msg, Interceptor.Flow flow, boolean includeJsonObject) {
+        HashMap<String, Object> parameters = new HashMap<>();
+        parameters.put("exc", exc);
+        parameters.put("flow", flow);
+
+        if (flow == REQUEST) {
+            try {
+                // TODO get URIFactory from Router
+                Map<String, String> queryParams = URLParamUtil.getParams(new URIFactory(),exc, URLParamUtil.DuplicateKeyOrInvalidFormStrategy.ERROR);
+                parameters.put("params",queryParams);
+            } catch (Exception e) {
+                log.info("Cannot parse query parameter from " + exc.getRequest().getUri());
+            }
+        }
+
+        if (msg != null) {
+            parameters.put("message", msg);
+            parameters.put("header", msg.getHeader());
+            parameters.put("body", msg.getBodyAsStringDecoded());
+            if (includeJsonObject) {
+                try {
+                    log.info("Parsing body as JSON for scripting plugins");
+                    parameters.put("json",om.readValue(readInputStream(msg.getBodyAsStream()),Map.class));  // @Todo not with Javascript
+                } catch (Exception e) {
+                    log.warn("Can't parse body as JSON: " + e);
```

This is bad practice, as it swallows the stacktrace of the exception. Please use `log.warn("Can't parse body as JSON: ", e);` instead.

---

Here the stacktrace do not provide more details. All the Info about the parsing is in the message. And we get a stacktrace in the Log everytime somebody sends a request containing invalid JSON.